controllers/grid/navigationMenus/NavigationMenusGridHandler.inc.php

```diff
@@ -0,0 +1,256 @@
+<?php
+
+/**
+ * @file controllers/grid/navigationMenus/NavigationMenusGridHandler.inc.php
+ *
+ * Copyright (c) 2014-2017 Simon Fraser University
+ * Copyright (c) 2003-2017 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class NavigationMenusGridHandler
+ * @ingroup controllers_grid_navigationMenus
+ *
+ * @brief Handle NavigationMenus grid requests.
+ */
+
+import('lib.pkp.classes.controllers.grid.GridHandler');
+import('lib.pkp.controllers.grid.navigationMenus.form.NavigationMenuForm');
+
+class NavigationMenusGridHandler extends GridHandler {
+	/**
+	 * Constructor
+	 */
+	function __construct() {
+		parent::__construct();
+		$this->addRoleAssignment(
+			ROLE_ID_MANAGER,
+			array(
+				'fetchGrid', 'fetchRow',
+				'addNavigationMenu', 'editNavigationMenu',
+				'updateNavigationMenu',
+				'deleteNavigationMenu'
+			)
+		);
+	}
+
+	//
+	// Overridden template methods
+	//
+	/**
+	 * @copydoc GridHandler::authorize()
+	 */
+	function authorize($request, &$args, $roleAssignments) {
+		//import('lib.pkp.classes.security.authorization.ContextAccessPolicy');
+		//$this->addPolicy(new ContextAccessPolicy($request, $roleAssignments));
+		$context = $request->getContext();
+
+		$contextId = CONTEXT_ID_NONE;
+		if ($context) {
+			$contextId = $context->getId();
+		}
+
+		$navigationMenuId = $request->getUserVar('navigationMenuId');
+		if ($navigationMenuId) {
+			// Ensure NavigationMenus is valid and for this context
+			$navigationMenuDao = DAORegistry::getDAO('NavigationMenuDAO'); /* @var $navigationMenuDao NavigationMenuDAO */
+			$navigationMenu = $navigationMenuDao->getById($navigationMenuId);
+			if (!$navigationMenu ||  $navigationMenu->getContextId() != $contextId) {
+				return false;
+			}
+		}
+		return parent::authorize($request, $args, $roleAssignments);
+	}
+
+	/**
+	 * @copydoc GridHandler::initialize()
+	 */
+	function initialize($request, $args = null) {
+		parent::initialize($request, $args);
+
+		// Basic grid configuration
+		$this->setTitle('manager.navigationMenus');
+
+		// Set the no items row text
+		$this->setEmptyRowText('grid.navigationMenus.navigationMenu.noneExist');
+
+		// Columns
+		import('lib.pkp.controllers.grid.navigationMenus.NavigationMenusGridCellProvider');
+		$navigationMenuCellProvider = new NavigationMenusGridCellProvider();
+
+		$this->addColumn(
+			new GridColumn('title',
+				'common.title',
+				null,
+				null,
+				$navigationMenuCellProvider
+			)
+		);
+
+		$this->addColumn(
+			new GridColumn('itemCount',
+				'common.count',
+				null,
+				null,
+				$navigationMenuCellProvider
+			)
+		);
+
+		$this->addColumn(
+			new GridColumn('default',
+				'common.default',
+				null,
+				null,
+				$navigationMenuCellProvider
+			)
+		);
```

Comment:
I think we could do away with the `itemCount` and `default` columns too. The default menus can all be edited to remove menu items anyway, so not sure there's any need to do anything special with them.

Reply:
OK